src/js/tls.js

```diff
@@ -205,21 +205,58 @@ function createServer(options, secureConnectionListener) {
   return new Server(options, secureConnectionListener);
 }
 
-function connect(options, callback) {
-  options = Object.create(options, {
-    isServer: { value: false, enumerable: true },
-  });
-
-  if (!options.host) {
-    options.host = 'localhost';
-  }
-  if (!options.port) {
-    options.port = 443;
+function connect(arg0, arg1, arg2, arg3) {
+  var options;
+  var tlsSocket;
+  var callback;
```

Comment:
I would call arg3 as callback. This would reduce the number of local variables. Also `callback = arg3` is unnecessary.